### PR TITLE
 follow up encounter feature

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
New follow up encounter feature
all is explained in the following pdf
[follow_up_encounter.pdf](https://github.com/openemr/openemr/files/4000243/follow_up_encounter.pdf)

### CODE REVIEW

File: sql/5_0_3-to-5_0_4_upgrade.sql [hunk 1]
```diff
@@ -0,0 +1,93 @@
+--
```
Comment (bradymiller) [inline sql/5_0_3-to-5_0_4_upgrade.sql]: This should be in sql/5_0_2-to-5_0_3_upgrade.sql

File: sql/5_0_3-to-5_0_4_upgrade.sql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+--  #IfInnoDBMigrationNeeded
+--    desc: find all MyISAM tables and convert them to InnoDB.
+--    arguments: none
+--    behavior: can take a long time.
+
+
+#IfMissingColumn form_encounter parent_encounter_id
+
+ALTER TABLE `form_encounter` ADD `parent_encounter_id` BIGINT(20) NULL DEFAULT NULL AFTER `pos_code`;
```
Comment (bradymiller) [inline sql/5_0_3-to-5_0_4_upgrade.sql]: remove the AFTER portion of this.

File: interface/patient_file/encounter/forms.php [hunk 3]
```diff
@@ -289,6 +289,22 @@ function deleteme() {
...
+function createFollowUpEncounter() {
+
+    <?php
+    $result = sqlQuery("SELECT * FROM form_encounter WHERE pid = ? AND encounter = ?", array(
+        $_SESSION['pid'],
+        $encounter
+    ));
+    $encounterId = (!empty($result['parent_encounter_id'])) ? $result['parent_encounter_id'] : $result['id'];
+    ?>
+    var data = {encounterId: '<?php echo $encounterId; ?>', mode: 'follow_up_encounter'};
```
Comment (bradymiller) [inline interface/patient_file/encounter/forms.php]: escape $encounterId with attr() :
```
var data = {encounterId: '<?php echo attr($encounterId); ?>', mode: 'follow_up_encounter'};
```

File: interface/patient_file/history/encounters.php [hunk 4]
```diff
@@ -796,10 +807,25 @@ function efmouseover(elem, ptid, encid, formname, formid) {
...
+
+    if ($GLOBALS['enable_group_therapy'] && !$billing_view && $therapy_group == 0) {
         $group_name = ($encounter_type['pc_cattype'] == 3 && is_numeric($result4['external_id'])) ? getGroup($result4['external_id'])['group_name']  : "";
         echo "<td>". text($group_name) . "</td>\n";
     }
 
+
+    if ($GLOBALS['enable_follow_up_encounters']) {
+        $encounterId= ( !empty($result4['parent_encounter_id']) ) ? $result4['parent_encounter_id'] : $result4['id'];
+        echo "<td> <div style='z-index: 9999'>  <a href='#' class='css_button' onclick='createFollowUpEncounter(event,".$encounterId.")'><span>".xlt('Create follow-up encounter')."</span></a> </div></td>\n";
```
Comment (bradymiller) [inline interface/patient_file/history/encounters.php]: escape $encounterId with attr_js() :
```
echo "<td> <div style='z-index: 9999'>  <a href='#' class='css_button' onclick='createFollowUpEncounter(event,".attr_js($encounterId).")'><span>".xlt('Create follow-up encounter')."</span></a> </div></td>\n";
```

**Comment by bradymiller** [comment]
hi @eyalvo-matrix ,
Looks like a useful feature for some workflows (agree with the global switch). Left some comments for minor issues. Once those are addressed, then will bring it in (I am assuming it's testing well for you). thanks! -brady

File: sql/5_0_3-to-5_0_4_upgrade.sql (hunk 1 above)
Comment (eyalvo-matrix) [reply sql/5_0_3-to-5_0_4_upgrade.sql]: ok will delete the new file

File: sql/5_0_3-to-5_0_4_upgrade.sql (hunk 2 above)
Comment (eyalvo-matrix) [reply sql/5_0_3-to-5_0_4_upgrade.sql]: sure , thanks

File: interface/patient_file/encounter/forms.php (hunk 3 above)
Comment (eyalvo-matrix) [reply interface/patient_file/encounter/forms.php]: Thanks :)

File: interface/patient_file/history/encounters.php (hunk 4 above)
Comment (eyalvo-matrix) [reply interface/patient_file/history/encounters.php]: Sure thanks :)

**Comment by eyalvo-matrix** [comment]
hi @bradymiller,  I have fixed all your comments (thanks on all of them) can you please have another look at it?

**Comment by bradymiller** [comment]
hi @eyalvo-matrix , Sorry for the delay. Will bring this into the codebase tomorrow. -brady

**Comment by bradymiller** [comment]
hi @eyalvo-matrix , code looks good and bringing into codebase. thanks for the new feature! -brady

**Pull request merged**